resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -258,32 +227,39 @@ public DnsNameResolver(
         this.ndots = checkPositiveOrZero(ndots, "ndots");
         this.decodeIdn = decodeIdn;
 
-        boolean supportsARecords = false;
-        boolean supportsAAAARecords = false;
-        // Use LinkedHashSet to maintain correct ordering.
-        Set<DnsRecordType> recordTypes = new LinkedHashSet<DnsRecordType>(resolvedAddressTypes.length);
-        for (InternetProtocolFamily family: resolvedAddressTypes) {
-            switch (family) {
-                case IPv4:
-                    supportsARecords = true;
-                    recordTypes.add(DnsRecordType.A);
-                    break;
-                case IPv6:
-                    supportsAAAARecords = true;
-                    recordTypes.add(DnsRecordType.AAAA);
-                    break;
-                default:
-                    throw new Error();
-            }
+        switch (resolvedAddressTypes) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) NullPointerException might be thrown as 'resolvedAddressTypes' is nullable here [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2259)